### PR TITLE
AV-158136: Validate enableSSL on listener Port only when listener settings are configured

### PR DESCRIPTION
**Pull request by akshayhavile**
…

### CODE REVIEW

**Comment by akshayhavile** [comment]
build ako

File: internal/k8s/crdcontroller.go [hunk 1]
```diff
@@ -698,7 +698,7 @@ func validateHostRuleObj(key string, hostrule *akov1alpha1.HostRule) error {
 		}
 	}
 
-	if hostrule.Spec.VirtualHost.TCPSettings != nil {
+	if hostrule.Spec.VirtualHost.TCPSettings != nil && len(hostrule.Spec.VirtualHost.TCPSettings.Listeners) > 0 {
```
Comment (swathinsankaran) [inline internal/k8s/crdcontroller.go:701]: Do we have to inform the user in such a scenario?

Comment (akshayhavile) [reply internal/k8s/crdcontroller.go:701]: @swathinsankaran `SSL enabled ` error should be raised only when listeners are specified. TCPSetting can have other setting without specifying listeners e.g. laodBalancerIP.

Comment (akshayhavile) [reply internal/k8s/crdcontroller.go:701]: So we do not require to notify anything to customer

**Review by swathinsankaran — APPROVED** [review]
LGTM

**Review by aaha97 — APPROVED** [review]

**Pull request merged**